beacon-chain/blockchain/service.go

```diff
@@ -236,133 +222,78 @@ func (c *ChainService) blockProcessing(done <-chan struct{}) {
 			return
 		// Listen for a newly received incoming block from the sync service.
 		case block := <-c.incomingBlockChan:
-			// 3 steps:
-			// - Compute the active state for the block.
-			// - Compute the crystallized state for the block if cycle transition.
-			// - Store both states and the block into a data structure used for fork choice.
-			//
-			// Another routine will run that will continually compute
-			// the canonical block and states from this data structure using the
-			// fork choice rule.
-			var canProcessBlock bool
-			var canProcessAttestations bool
-			var err error
-			var blockVoteCache map[[32]byte]*types.VoteCache
-
-			h, err := block.Hash()
-			if err != nil {
-				log.Debugf("Could not hash incoming block: %v", err)
-			}
-
-			receivedSlotNumber := block.SlotNumber()
-
-			log.WithField("blockHash", fmt.Sprintf("0x%x", h)).Info("Received full block, processing validity conditions")
-
-			parentExists, err := c.chain.hasBlock(block.ParentHash())
+			// 1. Validate the block
```

Comment:
Can we abstract any of this logic? For example, which bits of this routine are validating the block?

Reply:
Hmm I'm not sure what you mean. [This line](https://github.com/prysmaticlabs/prysm/pull/473/files/462ffaa5f3aaf3532cf35ff8c0379920cc1c3c00#diff-7f087f690e8ec9178851c9b6844cb529R246) is the entirety of block validation, so it's already fairly abstracted.

Reply:
What prompted me to say this is that I wasn't sure what it meant to validate the block in this context.